c7n/resources/iam.py

```diff
@@ -302,6 +302,21 @@ def _user_policies(resource):
         return matched
 
 
+@User.filter_registry.register('access-key-last-used')
+class UserAccessKeyLastUsed(Filter):
+    schema = type_schema(
+        'access-key-last-used',
+        keyid={'type': 'string'})
+
+    def process(self, resources, event=None):
+        keyid = self.data.get('keyid', None)
```

Comment:
the filter should be a match on users to find users with keys over older then a given date. the user resource  should be annotated with the key ids that match the filter.